### ISSUE TITLE
2.0: Native crash during replication. (ToDo sample)

### ISSUE THREAD

**Issue by hideki** [couchbase/couchbase-lite-android#1394]
```
09-11 16:03:50.409 21801-21801/com.couchbase.todo A/libc: /Volumes/Android/buildbot/src/android/ndk-r15-release/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:74: void abort_message(const char *, ...): assertion "terminating with uncaught exception of type litecore::error: bad revision ID" failed
09-11 16:03:50.410 21801-21801/com.couchbase.todo A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 21801 (.couchbase.todo)
                                                          
                                                          [ 09-11 16:03:50.412   271:  271 W/         ]
                                                          debuggerd: handling request: pid=21801 uid=10184 gid=10184 tid=21801
09-11 16:03:50.496 22438-22438/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
09-11 16:03:50.496 22438-22438/? A/DEBUG: Build fingerprint: 'google/shamu/shamu:7.1.1/N6F27I/4118426:user/release-keys'
09-11 16:03:50.496 22438-22438/? A/DEBUG: Revision: '0'
09-11 16:03:50.496 22438-22438/? A/DEBUG: ABI: 'arm'
09-11 16:03:50.497 22438-22438/? A/DEBUG: pid: 21801, tid: 21801, name: .couchbase.todo  >>> com.couchbase.todo <<<
09-11 16:03:50.497 22438-22438/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
09-11 16:03:50.511 22438-22438/? A/DEBUG: Abort message: '/Volumes/Android/buildbot/src/android/ndk-r15-release/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:74: void abort_message(const char *, ...): assertion "terminating with uncaught exception of type litecore::error: bad revision ID" failed'
09-11 16:03:50.511 22438-22438/? A/DEBUG:     r0 00000000  r1 00005529  r2 00000006  r3 00000008
09-11 16:03:50.511 22438-22438/? A/DEBUG:     r4 b2d3658c  r5 00000006  r6 b2d36534  r7 0000010c
09-11 16:03:50.511 22438-22438/? A/DEBUG:     r8 bea2a068  r9 b0085400  sl 00000000  fp bea29ff4
09-11 16:03:50.511 22438-22438/? A/DEBUG:     ip 0000000b  sp bea29a20  lr b0e3f5c7  pc b0e41e30  cpsr 600e0010
09-11 16:03:50.534 22438-22438/? A/DEBUG: backtrace:
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #00 pc 00049e30  /system/lib/libc.so (tgkill+12)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #01 pc 000475c3  /system/lib/libc.so (pthread_kill+34)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #02 pc 0001d635  /system/lib/libc.so (raise+10)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #03 pc 00019181  /system/lib/libc.so (__libc_android_abort+34)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #04 pc 00017048  /system/lib/libc.so (abort+4)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #05 pc 0001b633  /system/lib/libc.so (__libc_fatal+22)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #06 pc 0001937b  /system/lib/libc.so (__assert2+18)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #07 pc 0022323b  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #08 pc 00223329  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #09 pc 002212dd  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #10 pc 00221393  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so (_ZSt9terminatev+54)
09-11 16:03:50.534 22438-22438/? A/DEBUG:     #11 pc 000cee1b  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
```

**Comment by hideki** [comment]
REF #1393 

This could be caused by replicating conflicting document.

**Comment by hideki** [comment]
```
09-11 18:49:51.235 9403-9403/com.couchbase.todo E/HELLOW:   # 0: 0x92b8b112  _Z17backtraceToLogcatv
                                                            # 1: 0x92b8b83c  _ZN8litecore11revidBuffer5parseEN6fleece5sliceEb
                                                            # 2: 0x92b89486  _ZN10c4Internal12TreeDocument28selectCommonAncestorRevisionEN6fleece5sliceES2_
                                                            # 3: 0x92b9704c  c4doc_selectCommonAncestorRevision
                                                            # 4: 0x92b84f98  Java_com_couchbase_litecore_C4Document_selectCommonAncestorRevision
09-11 18:49:51.235 9403-9403/com.couchbase.todo E/BadRevisionID: revidBuffer::parse() - BadRevisionID
                                                                 
                                                                 --------- beginning of crash
09-11 18:49:51.237 9403-9403/com.couchbase.todo A/libc: /Volumes/Android/buildbot/src/android/ndk-r15-release/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:74: void abort_message(const char *, ...): assertion "terminating with uncaught exception of type litecore::error: bad revision ID" failed
09-11 18:49:51.238 9403-9403/com.couchbase.todo A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 9403 (.couchbase.todo)
                                                        
                                                        [ 09-11 18:49:51.239   271:  271 W/         ]
                                                        debuggerd: handling request: pid=9403 uid=10184 gid=10184 tid=9403
09-11 18:49:51.324 9744-9744/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
09-11 18:49:51.324 9744-9744/? A/DEBUG: Build fingerprint: 'google/shamu/shamu:7.1.1/N6F27I/4118426:user/release-keys'
09-11 18:49:51.324 9744-9744/? A/DEBUG: Revision: '0'
09-11 18:49:51.324 9744-9744/? A/DEBUG: ABI: 'arm'
09-11 18:49:51.325 9744-9744/? A/DEBUG: pid: 9403, tid: 9403, name: .couchbase.todo  >>> com.couchbase.todo <<<
09-11 18:49:51.325 9744-9744/? A/DEBUG: signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
09-11 18:49:51.351 9744-9744/? A/DEBUG: Abort message: '/Volumes/Android/buildbot/src/android/ndk-r15-release/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:74: void abort_message(const char *, ...): assertion "terminating with uncaught exception of type litecore::error: bad revision ID" failed'
09-11 18:49:51.351 9744-9744/? A/DEBUG:     r0 00000000  r1 000024bb  r2 00000006  r3 00000008
09-11 18:49:51.351 9744-9744/? A/DEBUG:     r4 b2d3658c  r5 00000006  r6 b2d36534  r7 0000010c
09-11 18:49:51.351 9744-9744/? A/DEBUG:     r8 bea2a068  r9 b0085400  sl 00000000  fp bea29ff4
09-11 18:49:51.351 9744-9744/? A/DEBUG:     ip 0000000b  sp bea29a20  lr b0e3f5c7  pc b0e41e30  cpsr 600f0010
09-11 18:49:51.379 9744-9744/? A/DEBUG: backtrace:
09-11 18:49:51.379 9744-9744/? A/DEBUG:     #00 pc 00049e30  /system/lib/libc.so (tgkill+12)
09-11 18:49:51.379 9744-9744/? A/DEBUG:     #01 pc 000475c3  /system/lib/libc.so (pthread_kill+34)
09-11 18:49:51.379 9744-9744/? A/DEBUG:     #02 pc 0001d635  /system/lib/libc.so (raise+10)
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #03 pc 00019181  /system/lib/libc.so (__libc_android_abort+34)
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #04 pc 00017048  /system/lib/libc.so (abort+4)
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #05 pc 0001b633  /system/lib/libc.so (__libc_fatal+22)
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #06 pc 0001937b  /system/lib/libc.so (__assert2+18)
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #07 pc 00223d1b  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #08 pc 00223e09  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #09 pc 00221dbd  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #10 pc 00221e73  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so (_ZSt9terminatev+54)
09-11 18:49:51.380 9744-9744/? A/DEBUG:     #11 pc 000cf333  /data/app/com.couchbase.todo-1/lib/arm/libLiteCoreJNI.so
                                        
                                        
                                        [ 09-11 18:49:52.711   271:  271 W/         ]
                                        debuggerd: resuming target 9403
```

**Comment by hideki** [comment]
The root cause of this problem is fixed with the ticket #1393 .

Filed the ticket for native issue to https://github.com/couchbase/couchbase-lite-core/issues/262

So I will close this.
